subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultFileCopyDetails.java

```diff
@@ -215,10 +216,15 @@ public ContentFilterable filter(Class<? extends FilterReader> filterType) {
 
     @Override
     public ContentFilterable expand(Map<String, ?> properties) {
-        filterChain.expand(properties);
+        filterChain.expand(mutablePropertiesCopy(properties));
         return this;
     }
 
+    private Map<String, ?> mutablePropertiesCopy(final Map<String, ?> properties) {
+        return new LinkedHashMap<>(properties);
+    }
+
```

Comment:
I don't think that this is the right place to fix this. This should be fixed where `FilterChain::expand` is implemented